api/client/src/main/java/org/projectnessie/client/auth/oauth2/AuthorizationCodeFlow.java

```diff
@@ -246,7 +247,32 @@ private static HttpServer createServer(int port, HttpHandler handler) {
       throw new UncheckedIOException(e);
     }
     server.createContext(CONTEXT_PATH, handler);
-    InetAddress local = InetAddress.getLoopbackAddress();
+    InetAddress localhost = null;
+    try {
+      for (var networkInterfaces = NetworkInterface.getNetworkInterfaces();
+          networkInterfaces.hasMoreElements(); ) {
+        var networkInterface = networkInterfaces.nextElement();
+        if (!networkInterface.isLoopback()) {
+          continue;
+        }
+        for (var inetAddresses = networkInterface.getInetAddresses();
+            inetAddresses.hasMoreElements(); ) {
+          var inetAddress = inetAddresses.nextElement();
+          if ("localhost".equals(inetAddress.getHostName())) {
+            LOGGER.debug("Considering inetAddress {} for the local http server", inetAddress);
+            localhost = inetAddress;
+          } else {
+            LOGGER.debug("Dismissing inetAddress {} for the local http server", inetAddress);
+          }
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    InetAddress local = localhost != null ? localhost : InetAddress.getLoopbackAddress();
+    if (local != localhost) {
+      LOGGER.debug("Using inetAddress {} for the local http server", local);
+    }
     try {
       server.bind(new InetSocketAddress(local, port), 0);
```

Comment:
Idea to explore: maybe bind to the wildcard address by using the `InetSocketAddress(int port)` constructor.

Reply:
Wildcard sounds like a good idea to me :)